File: app/src/main/java/no/nav/tilbakemeldingsmottak/consumer/joark/domain/OpprettJournalpostRequestTo.java
package no.nav.tilbakemeldingsmottak.consumer.joark.domain;

import lombok.Builder;
import lombok.Value;

import java.util.ArrayList;
import java.util.List;

@Value
@Builder
public class OpprettJournalpostRequestTo {

	private JournalpostType journalpostType;
	private AvsenderMottaker avsenderMottaker;
	private Bruker bruker;
	private String tema;
	private String behandlingstema;
	private String tittel;
	private String kanal;
	private String journalfoerendeEnhet;
	private String eksternReferanseId;
	private List<Tilleggsopplysning> tilleggsopplysninger = new ArrayList<>();
	private Sak sak;
	private List<Dokument> dokumenter = new ArrayList<>();

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/consumer/joark/domain/OpprettJournalpostRequestTo.kt
package no.nav.tilbakemeldingsmottak.consumer.joark.domain

// Swagger: https://dokarkiv.dev.intern.nav.no/swagger-ui/index.html#/journalpostapi/opprettJournalpost
// Java: https://github.com/navikt/dokarkiv/blob/master/journalpost/src/main/java/no/nav/dokarkiv/journalpost/v1/api/opprettjournalpost/OpprettJournalpostRequest.java
data class OpprettJournalpostRequestTo(
    val journalpostType: JournalpostType,
    val bruker: Bruker,
    val sak: Sak,
    val tema: String,
    val kanal: String,
    val journalfoerendeEnhet: String,
    val avsenderMottaker: AvsenderMottaker,
    val tittel: String,
    val dokumenter: List<Dokument> = listOf(),

    val behandlingstema: String? = null,
    val eksternReferanseId: String? = null,
    val tilleggsopplysninger: List<Tilleggsopplysning> = listOf(),
)
